src/dryad/id/ElementID.hpp

```diff
@@ -0,0 +1,288 @@
+      {  77, Entry{  77, "Ir", "Iridium"      , {} } },
+      {  78, Entry{  78, "Pt", "Platinum"     , {} } },
+      {  79, Entry{  79, "Au", "Gold"         , {} } },
+      {  80, Entry{  80, "Hg", "Mercury"      , {} } },
+      {  81, Entry{  81, "Tl", "Thallium"     , {} } },
+      {  82, Entry{  82, "Pb", "Lead"         , {} } },
+      {  83, Entry{  83, "Bi", "Bismuth"      , {} } },
+      {  84, Entry{  84, "Po", "Polonium"     , {} } },
+      {  85, Entry{  85, "At", "Astatine"     , {} } },
+      {  86, Entry{  86, "Rn", "Radon"        , {} } },
+      {  87, Entry{  87, "Fr", "Francium"     , {} } },
+      {  88, Entry{  88, "Ra", "Radium"       , {} } },
+      {  89, Entry{  89, "Ac", "Actinium"     , {} } },
+      {  90, Entry{  90, "Th", "Thorium"      , {} } },
+      {  91, Entry{  91, "Pa", "Protactinium" , {} } },
+      {  92, Entry{  92, "U" , "Uranium"      , {} } },
+      {  93, Entry{  93, "Np", "Neptunium"    , {} } },
+      {  94, Entry{  94, "Pu", "Plutonium"    , {} } },
+      {  95, Entry{  95, "Am", "Americium"    , {} } },
+      {  96, Entry{  96, "Cm", "Curium"       , {} } },
+      {  97, Entry{  97, "Bk", "Berkelium"    , {} } },
+      {  98, Entry{  98, "Cf", "Californium"  , {} } },
+      {  99, Entry{  99, "Es", "Einsteinium"  , {} } },
+      { 100, Entry{ 100, "Fm", "Fermium"      , {} } },
+      { 101, Entry{ 101, "Md", "Mendelevium"  , {} } },
+      { 102, Entry{ 102, "No", "Nobelium"     , {} } },
+      { 103, Entry{ 103, "Lr", "Lawrencium"   , {} } },
+      { 104, Entry{ 104, "Rf", "Rutherfordium", {} } },
+      { 105, Entry{ 105, "Db", "Dubnium"      , {} } },
+      { 106, Entry{ 106, "Sg", "Seaborgium"   , {} } },
+      { 107, Entry{ 107, "Bh", "Bohrium"      , {} } },
+      { 108, Entry{ 108, "Hs", "Hassium"      , {} } },
+      { 109, Entry{ 109, "Mt", "Meitnerium"   , {} } },
+      { 110, Entry{ 110, "Ds", "Darmstadtium" , {} } },
+      { 111, Entry{ 111, "Rg", "Roentgenium"  , {} } },
+      { 112, Entry{ 112, "Cn", "Copernicium"  , {} } },
+      { 113, Entry{ 113, "Nh", "Nihonium"     , {} } },
+      { 114, Entry{ 114, "Fl", "Flerovium"    , {} } },
+      { 115, Entry{ 115, "Mc", "Moscovium"    , {} } },
+      { 116, Entry{ 116, "Lv", "Livermorium"  , {} } },
+      { 117, Entry{ 117, "Ts", "Tennessine"   , {} } },
+      { 118, Entry{ 118, "Og", "Oganesson"    , {} } }
+    };
+    static inline const std::unordered_map< std::string, int > conversion_dictionary = [] ( const auto& dictionary ) {
+
+      std::unordered_map< std::string, int > conversion;
+      for ( const auto& [ number, entry ] : dictionary ) {
+
+        conversion[ entry.symbol() ] = number;
+        conversion[ entry.name() ] = number;
+        for ( const auto& alternative : entry.alternatives() ) {
+
+          conversion[ alternative ] = number;
+        }
+      }
+      return conversion;
+    }( element_dictionary );
+
+    /* fields */
+    const Entry* entry_;

```

Comment:
I know all of these are const pointers to static memory, so this all should be safe, but this seems like massive overkill to save a few bytes.

Reply:
It is basically a trial implementation, to see if it would work for other identifiers. Reaction identifiers would contain a list of outgoing particles, the equivalent MT number if it has one, etc.